Meta/run.sh

```diff
@@ -47,19 +61,30 @@ $SERENITY_EXTRA_QEMU_ARGS
 
 export SDL_VIDEO_X11_DGAMOUSE=0
 
+[ ! "${SERENITY_BUILD+is_set}" ] && {
```

Comment:
> I believe the correct way to to specify an alternative value is with :+ instead of just +. That said, this bit would be clearer with just if [ -z "$SERENITY_BUILD" ]; then

Why are you saying this? that is just wrong:
It is true that `[ ! "${SERENITY_BUILD:+is_set}" ]` would be the same as `[ -z  "$SERENITY_BUILD" ]`, but I'm using `+`, not `:+`...

The difference is what happens when "the variable is set, but empty", example:
```sh
#!/bin/sh
unset VARIABLE; echo unset VARIABLE
[ ! "${VARIABLE:+is_set}" ] && echo '[ ! "${VARIABLE:+is_set}" ]'
[ !  "${VARIABLE+is_set}" ] && echo '[ !  "${VARIABLE+is_set}" ]'
[ -z          "$VARIABLE" ] && echo '[ -z          "$VARIABLE" ]'
VARIABLE=; echo VARIABLE=
[ ! "${VARIABLE:+is_set}" ] && echo '[ ! "${VARIABLE:+is_set}" ]'
[ !  "${VARIABLE+is_set}" ] && echo '[ !  "${VARIABLE+is_set}" ]'
[ -z          "$VARIABLE" ] && echo '[ -z          "$VARIABLE" ]'
VARIABLE=hello; echo VARIABLE=hello
[ ! "${VARIABLE:+is_set}" ] && echo '[ ! "${VARIABLE:+is_set}" ]'
[ !  "${VARIABLE+is_set}" ] && echo '[ !  "${VARIABLE+is_set}" ]'
[ -z          "$VARIABLE" ] && echo '[ -z          "$VARIABLE" ]'
```
 The output of this script will be:
```
unset VARIABLE
[ ! "${VARIABLE:+is_set}" ]
[ !  "${VARIABLE+is_set}" ]
[ -z          "$VARIABLE" ]
VARIABLE=
[ ! "${VARIABLE:+is_set}" ]
[ -z          "$VARIABLE" ]
VARIABLE=hello
```
As you can see, `[ !  "${VARIABLE+is_set}" ]` is only true when the variable `VARIABLE` is not set while `[ -z          "$VARIABLE" ]` and `[ ! "${VARIABLE:+is_set}" ]` are also true when VARIABLE is set, but empty.

If you run the `run.sh` without setting SERENITY_BUILD, this block of code will be executed and SERENITY_BUILD will be set to either `SERENITY_ROOT/Build` or `SERENITY_ROOT/build`.
If you run the script, setting SERENITY_BUILD to an empty value (i.e. `SERENITY_BUILD= Meta/run.sh`), this code will not be executed and SERENITY_BUILD will be set to `.` at line 71.

Reply:
> source: https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html

Your own source explains the difference between `:+` and `+`. :^)

> When not performing substring expansion, using the form described below (e.g., ‘:-’), Bash tests for a parameter that is unset or null. Omitting the colon results in a test only for a parameter that is unset.

Reply:
Thanks for the example script, I didn't know that expansion was possible. I can't find documentation for it, though.

> Your own source explains the difference between :+ and +. :^)

Searching for `+` in it gives me a single result, which is in `${name:+word}`.

> If you run the script, setting SERENITY_BUILD to an empty value (i.e. SERENITY_BUILD= Meta/run.sh),

This looks like a weird use case, but I don't know the full requirements for it. I see your point, though.

> Note that the documentation you posted is for the bash shell; this is not a bash script, but these parameter expansions also work in POSIX sh.

I am aware. I couldn't find a POSIX shell specification, and between posting the bash manual and a link to dash's man page, I thought the Bash link had a better look.

> > I believe the correct way to to specify an alternative value is with :+ instead of just +.

> Also, are you confusing the + operator with the - operator?

The dash manual calls `${parameter:+word}` an alternative value.

Reply:
> Thanks for the example script, I didn't know that expansion was possible. I can't find documentation for it, though.

I was refering to this paragraph.
> When not performing substring expansion, using the form described below (e.g., ‘:-’), Bash tests for a parameter that is unset or null. Omitting the colon results in a test only for a parameter that is unset. Put another way, if the colon is included, the operator tests for both parameter’s existence and that its value is not null; if the colon is omitted, the operator tests only for existence.

It basically says "when omitting the colon, the operator will only test if the variable is unset, instead of testing if the variable is unset or null".

i.e.
```man
${parameter:-word}
If parameter is unset or null, the expansion of word is substituted. Otherwise, the value of parameter is substituted.
```
without the colon becomes:
```man
${parameter-word}
If parameter is unset, the expansion of word is substituted. Otherwise, the value of parameter is substituted.
```

NOTE: "null" is how you call a variable which is set to an empty string.

> I couldn't find a POSIX shell specification

I usually use this website for reading POSIX specs (https://pubs.opengroup.org/onlinepubs/9699919799/).
Here is the spec for POSIX shell parameter expansions: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02

Reply:
I see, thank you very much for the explanations.